### PR TITLE
Sudoku

### PR DESCRIPTION
**Pull request by ishi0301**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes

### Description
Sudoku Solver

### Checklist

- [x ] I've read the contribution guidelines.
- [x ] I've checked the issue list before deciding what to submit.
- [x ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#976

### CODE REVIEW

**Comment by ishi0301** [comment]
@HarshCasper @Abhijit2505 @SKAUL05 @iamrajiv Please review my code.

**Comment by ishi0301** [comment]
Kindly review @iamrajiv @HarshCasper @siddharth25pandey @SKAUL05 @Kajol-Kumari @Abhijit2505 @plazzy99  .

**Comment by siddharth25pandey** [comment]
@ishi0301 Please remove the unnecessary **css & html** file

**Comment by ishi0301** [comment]
Removed unnecessary files. Please approve. @iamrajiv @HarshCasper @siddharth25pandey @SKAUL05 @Kajol-Kumari @Abhijit2505 @plazzy99.

File: C-Plus-Plus/other/sudoku_solver.cpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
+// SudokuSolver
+//Given a partially filled 9�9 2D array �grid[9][9]�, the goal is to assign digits (from 1 to 9) to the empty cells so that every row, column, and subgrid of size 3�3 contains exactly one instance of the digits from 1 to 9.
+#include <iostream>
+#include <cmath>
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/sudoku_solver.cpp]: Add single header file **bits/stdc++.h**

File: C-Plus-Plus/other/sudoku_solver.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+			}
+			mat[i][j]=0;
+		}
+	}
+	return false;
+}
+
+
+int main(){
+	int mat[9][9] =
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/sudoku_solver.cpp]: Take Dynamic Input from the user

File: C-Plus-Plus/other/sudoku_solver.cpp [hunk 3]
```diff
@@ -0,0 +1,109 @@
+// SudokuSolver
+//Given a partially filled 9�9 2D array �grid[9][9]�, the goal is to assign digits (from 1 to 9) to the empty cells so that every row, column, and subgrid of size 3�3 contains exactly one instance of the digits from 1 to 9.
+#include <iostream>
+#include <cmath>
+using namespace std;
+
+bool isSafeToPut(int mat[][9],int i,int j,int n,int number){
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/sudoku_solver.cpp]: Don't fix any number of rows/columns

File: C-Plus-Plus/other/sudoku_solver.cpp [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+				return false;
+			}
+		}
+	}
+	return true;
+
+}
+
+
+bool SudokuSolver(int mat[][9],int i,int j,int n){
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/sudoku_solver.cpp]: Don't fix any number of rows/columns

File: C-Plus-Plus/other/sudoku_solver.cpp (hunk 2 above)
Comment (siddharth25pandey) [inline C-Plus-Plus/other/sudoku_solver.cpp]: take rows & columns by user's input.

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@ishi0301 Please do the suggested changes
Thanks!

**Comment by ishi0301** [comment]
Made necessary changes. Please approve. @iamrajiv @HarshCasper @siddharth25pandey @SKAUL05 @Kajol-Kumari @Abhijit2505 @plazzy99.

File: C-Plus-Plus/other/sudoku_solver.cpp [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    }
+    return false;
+}
+
+int main()
+{
+    int n;
+    cout << "Enter size of square matrix: ";
+    cin >> n;
+    int mat[10][10];
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/sudoku_solver.cpp]: Don't fix the size of the grid

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@ishi0301 Don't directly fix the size of the grid.
What happen if user enter a square grid which is greater than 10, let's say 15*15.
Therefore, please change it.

**Comment by ishi0301** [comment]
@siddharth25pandey please check now and approve it. @iamrajiv @HarshCasper  @SKAUL05 @Kajol-Kumari @Abhijit2505 @plazzy99.

**Review by siddharth25pandey — APPROVED** [review]
@ishi0301 Good to go
Able to merge
Thanks!

**Review by iamrajiv — APPROVED** [review]

**Pull request merged**